src/System.Private.Uri/src/System/IPv4AddressHelper.cs

```diff
@@ -27,7 +27,18 @@ internal static string ParseCanonicalName(string str, int start, int end, ref bo
             {
                 byte* numbers = stackalloc byte[NumberOfLabels];
                 isLoopback = Parse(str, numbers, start, end);
-                return numbers[0] + "." + numbers[1] + "." + numbers[2] + "." + numbers[3];
+
+                Span<char> stackSpace = stackalloc char[NumberOfLabels * 3 + 3];
+                int totalChars = 0, charsWritten;
+                for (int i = 0; i < 3; i++)
+                {
+                    numbers[i].TryFormat(stackSpace.Slice(totalChars), out charsWritten);
```

Comment:
Is it worth it to loop unroll this to avoid this slice? It doesn't look like it is.

This doesn't get us much:
```C#
            unsafe
            {
                byte* numbers = stackalloc byte[NumberOfLabels];
                isLoopback = Parse(str, numbers, start, end);

                Span<char> stackSpace = stackalloc char[NumberOfLabels * 3 + 3];
                int totalChars = 0;

                numbers[0].TryFormat(stackSpace, out int charsWritten);
                totalChars += charsWritten + 1;
                stackSpace[totalChars - 1] = '.';

                numbers[1].TryFormat(stackSpace.Slice(totalChars), out charsWritten);
                totalChars += charsWritten + 1;
                stackSpace[totalChars - 1] = '.';

                numbers[2].TryFormat(stackSpace.Slice(totalChars), out charsWritten);
                totalChars += charsWritten + 1;
                stackSpace[totalChars - 1] = '.';

                numbers[3].TryFormat(stackSpace.Slice(totalChars), out charsWritten);

                return new string(stackSpace.Slice(0, totalChars + charsWritten));
            }
```

Reply:
Not worth it. Bigger and unsafe code.